internal/service/dhcp/filterrelayagent_resource_test.go

```diff
@@ -0,0 +1,782 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_name", circuitIdName),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccFilterrelayagentCircuitIdName(name, updateCircuitIdName, isCircuitId, isCircuitIdSubstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_name", updateCircuitIdName),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccFilterrelayagentResource_CircuitIdSubstringLength(t *testing.T) {
+	var resourceName = "nios_dhcp_filterrelayagent.test_circuit_id_substring_length"
+	var v dhcp.Filterrelayagent
+	name := acctest.RandomNameWithPrefix("filterrelayagent")
+	isCircuitID := "MATCHES_VALUE"
+	isCircuitIdSubstring := "true"
+	circuitIdSubstringOffset := "0"
+	circuitIdName := "CIRCUIT_ID_NAME_TEST"
+	circuitIdSubstringLength := fmt.Sprintf("%d", len(circuitIdName))
+	updatedCircuitIDName := "CIRCUIT_ID_NAME_TEST_UPDATED"
+	updatedCircuitIdSubstringLength := fmt.Sprintf("%d", len(updatedCircuitIDName))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccFilterrelayagentCircuitIdSubstringLength(name, isCircuitID, circuitIdName, circuitIdSubstringLength, circuitIdSubstringOffset, isCircuitIdSubstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_substring_length", circuitIdSubstringLength),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccFilterrelayagentCircuitIdSubstringLength(name, isCircuitID, updatedCircuitIDName, updatedCircuitIdSubstringLength, circuitIdSubstringOffset, isCircuitIdSubstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_substring_length", updatedCircuitIdSubstringLength),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccFilterrelayagentResource_CircuitIdSubstringOffset(t *testing.T) {
+	var resourceName = "nios_dhcp_filterrelayagent.test_circuit_id_substring_offset"
+	var v dhcp.Filterrelayagent
+	name := acctest.RandomNameWithPrefix("filterrelayagent")
+	isCircuitID := "MATCHES_VALUE"
+	isCircuitIdSubstring := "true"
+	circuitIdSubstringOffset := "0"
+	circuitIdName := "CIRCUIT_ID_NAME_TEST"
+	isCircuitIdSubstringLength := fmt.Sprintf("%d", len(circuitIdName))
```

Comment:
```suggestion
	circuitIdSubstringLength := fmt.Sprintf("%d", len(circuitIdName))
```
Please rename it as these are int variables